File: lab3/Java/src/business/service/IBookService.java
package business.service;

import business.model.BookModel;

import java.util.List;


/**
 * Created by Alex on 07/03/2017.
 * Adapted by Andrei 16/06/2018.
 */
public interface IBookService {

    List<BookModel> findAll();

    BookModel findById(Long id);

    boolean create(BookModel book);

    int getAgeOfBook(Long id);

}

File: lab3/C#/Layers/BusinessLogic/Contracts/IBookService.cs
using BusinessLogic.Model;

namespace BusinessLogic.Contracts
{
    public interface IBookService
    {
        BookModel GetById(int id);
        BookModel[] GetAll();

        bool Add(BookModel book);
        void Update(BookModel book);
        void Delete(int id);

        int GetAgeOfBook(int id);
    }
}
